.circleci/config.yml

```diff
@@ -16,7 +16,502 @@ aliases:
+
+  build-oss-frontend:
+    executor: grafana-build
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: Build Grafana frontend
+          command: /tmp/workspace/bin/grabpl build-frontend
+      - run:
+          name: Move artifacts
+          command: mkdir oss && cp -r public oss/
+      - persist_to_workspace:
+          root: .
+          paths:
+            - oss/public/*
+      - run:
+          name: CI job failed
+          command: ./scripts/ci-job-failed.sh
+          when: on_fail
+      - run:
+          name: CI job succeeded
+          command: ./scripts/ci-job-succeeded.sh
+          when: on_success
+
+  build-enterprise-backend:
+    description: "Build a certain variant of Grafana enterprise back-end binaries"
+    parameters:
+      variant:
+        type: string
+    executor: grafana-build
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: Build Grafana backend
+          # TODO: Remove $CIRCLE_TAG for master
+          command: /tmp/workspace/bin/grabpl build-backend --enterprise --variants << parameters.variant >> $CIRCLE_TAG
+      - run:
+          name: Move artifacts
+          command: mkdir enterprise && mv bin enterprise/
+      - persist_to_workspace:
+          root: .
+          paths:
+            - enterprise/bin/*
+      - run:
+          name: CI job failed
+          command: ./scripts/ci-job-failed.sh
+          when: on_fail
+      - run:
+          name: CI job succeeded
+          command: ./scripts/ci-job-succeeded.sh
+          when: on_success
+
+  build-enterprise-frontend:
+    executor: grafana-build
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: Build Grafana frontend
+          command: /tmp/workspace/bin/grabpl build-frontend --enterprise
+      - run:
+          name: Move artifacts
+          command: mkdir enterprise && cp -r public enterprise/
+      - persist_to_workspace:
+          root: .
+          paths:
+            - enterprise/public/*
+      - run:
+          name: CI job failed
+          command: ./scripts/ci-job-failed.sh
+          when: on_fail
+      - run:
+          name: CI job succeeded
+          command: ./scripts/ci-job-succeeded.sh
+          when: on_success
+
+  shellcheck-grabpl:
+    machine: true
+    steps:
+      - attach_workspace:
+          at: /tmp/workspace
+      - checkout
+      - run:
+          name: CI job started
+          command: ./scripts/ci-job-started.sh
+      - run:
+          name: ShellCheck
+          command: /tmp/workspace/bin/grabpl shellcheck
```

Comment:
what does this do?

Reply:
@torkelo This centralizes some build logic in the build-pipeline tool: https://github.com/grafana/build-pipeline/blob/master/pkg/subcmd/shellcheck/action.go#L14.

Reply:
Nit. Would make shellcheck more general by require `"-e", "SC1071", "-e", "SC2162"` to be provided by arguments/config instead of hardcoded in the code.

Reply:
@marefr Is there any need currently for this generalization? My program doesn't differ any from the current Makefile in the respect that it codifies our configuration of shellcheck. My thinking is that once we need different shellcheck configurations, we make it configurable.

Reply:
No it's not important for now. Was just a nit/something to think about moving forward.